frontends/election-manager/src/utils/generate_results_csv.ts

```diff
@@ -0,0 +1,177 @@
+import { filterTalliesByParams } from '@votingworks/utils';
+import {
+  Election,
+  expandEitherNeitherContests,
+  writeInCandidate,
+  FullElectionTally,
+  VotingMethod,
+  Precinct,
+  Contest,
+} from '@votingworks/types';
+
+function buildCsvRow(
+  precinct: Precinct,
+  votingMethod: string,
+  votes: number | undefined,
+  contest: Contest,
+  selection: string,
+  selectionId?: string
+): string {
+  return [
+    `"${precinct.name}"`,
+    precinct.id,
+    votingMethod,
+    votes?.toString() ?? '0',
+    `"${contest.title}"`,
+    contest.id,
+    `"${selection}"`,
+    selectionId ?? '',
+  ].join(',');
+}
```

Comment:
feels like we should be using an existing `csv` library so we don't have to do our own escaping, and we handle various character encoding edge cases automatically.